peer/brontide.go

```diff
@@ -3796,6 +3799,8 @@ func (p *Brontide) initRbfChanCloser(
 	// Finally, we'll register this new endpoint with the message router so
 	// future co-op close messages are handled by this state machine.
 	err = fn.MapOptionZ(p.msgRouter, func(r msgmux.Router) error {
+		_ = r.UnregisterEndpoint(chanCloser.Name())
```

Comment:
Is this necessary?

Reply:
This makes sure that we wipe the state of the endpoint before adding a new one. Otherwise, we'll get an error. This can happen if we get an error (`ReportError`), so we want to wipe the protocol state and create a new chan closer.